tests/test_util.py

```diff
@@ -222,12 +222,19 @@ async def asyncSetUp(self):
     async def asyncTearDown(self):
         await self.http.close()
 
+    async def test_correct_content_type(self):
+        await get_extra_data_info_from_url(
+            url=f"https://ftp.gnu.org/gnu/gzip/gzip-1.12.tar.gz",
+            session=self.http,
+            content_type_deny=[re.compile(f"^{self._CONTENT_TYPE}$")],
+        )
+
     async def test_wrong_content_type(self):
```

Comment:
It's kind of weird that the only test of the new behaviour – ignoring the `Content-Type` header and instead guessing from the first chunk of data – is here, in a kind of double-negative way.

I had to refresh my memory of what this `content_type_deny` feature is for, and it's for Sourceforge which returns HTML pages with a 200 OK status code rather than source code archives in some scenarios.

OK, I guess this works. But can we have a comment explaining it, something like  (line-wrapped as appropriate):

```suggestion
    async def test_wrong_content_type(self):
        # Because so many web servers serve source code archives with an incorrect
        # Content-Type, we ignore the Content-Type header and instead use
        # (lib)magic to sniff the content type from the response data.
        #
        # However, we also have to deal with the problem that SourceForge
        # sometimes returns a 200 OK response with an HTML error page rather
        # than the source code archive we requested. So
        # get_extra_data_info_from_url() allows the caller to provide a
        # denylist.
        #
        # This test case is testing 2 things. The server returns a JSON body
        # but with application/gzip as the Content-Type. We test that:
        #
        # 1. The content type is sniffed from the data as application/json
        # 2. As a result it is rejected
```

Reply:
Yes, it's a bit confusing. I've added your explanation.